docs/app-plan.doc

```diff
@@ -0,0 +1,173 @@
+* Create a `disabled` application plan by `--disabled` flag. By default, it will be `enabled`.
+* Several other options can be set. Check `usage`
+
+```shell
+NAME
+    create - create application plan
+
+USAGE
+    3scale application-plan create [opts] <remote>
+    <service> <plan_name>
+
+DESCRIPTION
+    Create application plan
+
+OPTIONS
+       --approval-required=<value>      Applications require approval. true
+                                        or false
+       --cost-per-month=<value>         Cost per month
+    -d --default                        Make default application plan
+       --disabled                       Disable application plan
+       --end-user-required=<value>      End user required. true or false
+    -p --published                      Publish application plan
+       --setup-fee=<value>              Setup fee
+    -t --system-name=<value>            Application plan system name
+       --trial-period-days=<value>      Trial period days
+
+OPTIONS FOR APPLICATION-PLAN
+    -c --config-file=<value>            3scale toolbox configuration file
+                                        (default:
+                                        $HOME/.3scalerc.yaml)
+    -h --help                           show help for this command
+    -k --insecure                       Proceed and operate even for server
+                                        connections otherwise considered
+                                        insecure
+    -v --version                        Prints the version of this command
+       --verbose                        Verbose mode
+```
+
+### Apply
+
+* Update existing application plan. Create new one if it does not exist.
+* `service` positional argument is a service reference. It can be either service `id`, or service `system_name`. Toolbox will figure it out.
+* `plan` positional argument is a plan reference. It can be either plan `id`, or plan `system_name`. Toolbox will figure it out.
+* This is command is `idempotent`.
+* Update to `default` application plan by `--default` flag.
+* Update to `published` application plan by `--publish` flag.
+* Update to `hidden` application plan by `--hide` flag.
+* Update to `disabled` application plan by `--disabled` flag.
+* Update to `enabled` application plan by `--enabled` flag.
+* Several other options can be set. Check `usage`
+
+```shell
+NAME
+    apply - Update application plan
+
+USAGE
+    3scale application-plan apply [opts] <remote>
+    <service> <plan>
+
+DESCRIPTION
+    Update (create if it does not exist) application plan
+
+OPTIONS
+       --approval-required=<value>      Applications require approval. true
+                                        or false
+       --cost-per-month=<value>         Cost per month
+       --default                        Make default application plan
+       --disabled                       Disable application plan
+       --enabled                        Enable application plan
```

Comment:
I wonder if this is needed. It would remove all 0/eternity limits I guess?

I was expecting a boolean: disabled: true false

Reply:
This is `apply` command, like `update`. The default behavior should be only update things I tell you to update. If we implement `--disabled` default value as `false`, hence enabled, updating disabled app plan to change only, for instance  the `cost-per-month`, would enable as well the application plan, deleting all eternity zero metrics.

For the `create` command, it makes sense your expectation and that is the behavior it was implemented for the `create` command.

IMO, for the `apply` command, you should explicitly state you want the plan to be `enabled` or `disabled` or do not change current status.

The same principle applies for `approval-required` and `end-user-required`. This is why, they are not `flags`, but `options` with required value. If not set, `apply` will not change their current state. And if you want to update them, you must provide a value, either `true` or `false`

WDYT?? make this sense to you?